src/NUnitFramework/framework/TestContext.cs

```diff
@@ -417,6 +417,24 @@ public PropertyBagAdapter Properties
                 get { return _test.Arguments; }
             }
 
+            /// <summary>
+            /// The Type of the test.
+            /// Returns null if no fixture type is associated with it.
+            /// </summary>
+            public Type? TypeInfo
+            {
+                get { return _test.TypeInfo?.Type; }
+            }
+
+            /// <summary>
+            /// The MethodInfo of the test.
+            /// Returns null if the test is not implemented as a method
+            /// </summary>
+            public MethodInfo? MethodInfo
```

Comment:
Same question here. I see you're using this [in your code](https://github.com/VerifyTests/Verify/blob/19983c954581557dd7fea0df3b489a6c4083fcce/src/Verify.NUnit/Verifier.cs#L22) to pass to TestNameBuilder.GetUniqueTestName. It sounds like a more direct feature for NUnit to expose might be a guaranteed unique test name property which might address a more generalizable need.

Reply:
I am not sure if Verifies unique test name would be useful to Nunit? it is currently https://github.com/VerifyTests/Verify/blob/master/src/Shared/TestNameBuilder.cs

but either way, people can extend nunit in many ways, and exposing the current type and method info enables those scenarios

Reply:
Do you need that specific implementation, or could you also find it useful if we exposed something similar to VSTest's FQN (fully-qualified name)? Not saying we should do that, but just as an example.

Reply:
> Do you need that specific implementation 

yep

exposing an FQN for the would probably work. but i need the full methodinfo, so it would seem strange to expose the methodinfo and not the typeinfo.

Reply:
The FQN would cover both the type and method name. Is there something that you need the full methodinfo for that the FQN (or some guaranteed unique name) doesn't give you?

Reply:
>  Is there something that you need the full methodinfo for that the FQN (or some guaranteed unique name) doesn't give you?

the parameter infos.

Reply:
VSTest's FQN includes the parameter names for the method as well as (IIRC) parameter values. Assuming you had a test name like this: `A.B.C.Method(x: 1, y: "2")`, would you still need access to the System.Reflection side of the test?

Reply:
a test name like that wont help me. i would still need to reflect into the internals of nunit

Reply:
Continuing at https://github.com/nunit/nunit/pull/3557#issuecomment-731905892